lib/vm/src/instance/mod.rs

```diff
@@ -888,14 +856,21 @@ impl Instance {
         //}
 
         let ret = unsafe { memory32_atomic_check32(memory, dst, val) };
-
         if let Ok(mut ret) = ret {
             if ret == 0 {
-                ret = self.do_wait(memory_index.as_u32(), dst, timeout);
-            }
-            if ret == 0xffff {
-                // ret is 0xffff if there is more than 2^32 waiter in queue
-                return Err(Trap::lib(TrapCode::TableAccessOutOfBounds));
+                let memory = self.get_vmmemory_mut(memory_index);
+                let location = NotifyLocation { address: dst };
```

Comment:
This is quite a lot of duplicated code for local/non local.

We should only duplicate the code for retrieving the memory, the other logic should be in a common function.